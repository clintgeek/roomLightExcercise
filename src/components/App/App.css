.container {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: grid;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  /* grid-template-rows: repeat(3, 1fr); */
  grid-template-rows: 1fr 2rem;
  grid-template-columns: repeat(3, 1fr);
  grid-template-areas: 
    ". top-center ."
    ". . bottom-right";
}

.encouragement {
  padding-top: .5rem;
  color: darkgray;
  font-size: medium;
}

.room {
  grid-area: top-center;
}

.moreInfo {
  color: darkgray;
  font-size: calc(10px + 1vmin);
  grid-area: bottom-right;
  justify-self: end;
  padding: 0 .5rem .5rem 0;
}

.link {
  color: darkgray;
}

.link:hover {
  color: lightgray;
}