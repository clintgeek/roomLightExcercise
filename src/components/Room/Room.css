.roomName {
  padding-bottom: 1rem;
}

.walledRoom {
  border: 1px solid darkgray;
  padding: 20px;
  background: #222222;
}

.bright {
  border: 1px solid black;
  background: lightgray;
}