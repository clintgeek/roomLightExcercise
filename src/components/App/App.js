import React from 'react';
import './App.css';
import Room from '../Room';

function App() {
  return (
    <div className="container">
      <div className="room">
        <Room name="Test Room" />
      </div>
      <div className="moreInfo">
        <a className="link" href="https://github.com/clintgeek/roomLightExcercise">github.com/clintgeek/roomLightExcercise</a>
      </div>
    </div>
  );
}

export default App;
