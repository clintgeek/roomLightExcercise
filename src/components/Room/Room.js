import React, { useState } from 'react';
import LightBulb from '../LightBulb/';
import LightSwitch from '../LightSwitch';
import './Room.css';

function Room(props) {
  const [ powered, setPower ] = useState(false);

  const togglePower = () => setPower(!powered);

  var lightLevelClass = powered ? ' bright' : '';
  const roomClass = 'walledRoom' + lightLevelClass;

  return (
    <div>
      <div className="roomName">{props.name}</div>
      <div className={roomClass}>
        <LightBulb powered={powered}/>
        <LightSwitch powered={powered} togglePower={togglePower} />
      </div>
    </div>
  );
}

export default Room;