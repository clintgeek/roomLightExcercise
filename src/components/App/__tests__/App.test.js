import React from 'react';
import {shallow} from 'enzyme';
import App from '../App';
import Room from '../../Room';

describe('<App> component', () => {
  const wrapper = shallow(<App />);

  it('should match snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })

  it('should render <Room>', () => {
    const room = wrapper.find(Room);
    expect(room.isEmptyRender()).toBe(false);
  });
});

