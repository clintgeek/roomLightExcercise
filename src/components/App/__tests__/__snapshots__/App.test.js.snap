// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> component should match snapshot 1`] = `
"<div className=\\"container\\">
  <div className=\\"room\\">
    <Room name=\\"Test Room\\" />
  </div>
  <div className=\\"moreInfo\\">
    <a className=\\"link\\" href=\\"https://github.com/clintgeek/roomLightExcercise\\">
      github.com/clintgeek/roomLightExcercise
    </a>
  </div>
</div>"
`;
